guides/installing-python.md

```diff
@@ -0,0 +1,147 @@
+brew cleanup python
+brew uninstall virtualenvwrapper
+```
+You should also remove any variables that reference Homebrew's Python from your shell's dotfiles (`.profile`, `.bashrc` for bash users, `.zshrc` for zsh users), for example:
+
+```shell
+export PYTHONPATH=$HOME/homebrew/bin/python
+```
+
+If you do not wish to uninstall Homebrew's Python, or if you have Homebrew-installed packages that depend on Python-installed Homebrew, simply ensure that Homebrew's Python either does not exist in your shell's `PATH` variable, or that it is added to `PATH` *before* the pyenv configuration below.
+
+## Installing pyenv
+
+To [install pyenv on MacOS](https://github.com/pyenv/pyenv#homebrew-on-macos), use Homebrew:
+
+```shell
+brew update
+brew install pyenv
+brew install pyenv-virtualenv
+brew install pyenv-virtualenvwrapper
+```
+
+**Note:** pyenv-virtualenvwrapper is not strictly required, depending on how you wish to manage your virtualenvs. See the [notes in the pyenv-virtualewrapper README](https://github.com/pyenv/pyenv-virtualenvwrapper#pyenv-virtualenvwrapper). Some of our documentation assumes `virtualenvwrapper` exists, so this guide installs it. You will see a deprecation warning when using `mkvirtualenv`:
+
+```
+WARNING: the pyenv script is deprecated in favour of `python3.6 -m venv`
+```
+
+Until we switch to using Python 3 and the `venv` module full-time, this can be safely ignored.
+
+## Configuring pyenv
+
+Per the [pyenv documentation](https://github.com/pyenv/pyenv#basic-github-checkout), the follow environment variables need to be added to your shell dotfiles (`.bashrc` for bash users, `.zshrc` for zsh users).
+
+```shell
+# pyenv
+export PYENV_ROOT="$HOME/.pyenv"
+
+# pyenv-virtualenvwrapper
+export PYENV_VIRTUALENVWRAPPER_PREFER_PYVENV="true"
+export WORKON_HOME=$HOME/.virtualenvs
+
+# Ensure this placed toward the end of the file
+eval "$(pyenv init -)"
+
+# Load virtualenvwrapper
+pyenv virtualenvwrapper_lazy
+```
+
+Restart your shell.
+
+## Installing Python
+
+We have projects that require both Python 2.7 and Python 3.6:
+
+```shell
+pyenv install 3.6.8
+pyenv install 2.7.15
+```
+
+With these two versions of Python installed, you 
+
+## Setting the global Python versions
+
+By default with pyenv when you invoke `python`, you will be using the system-installed Python. To set the [global Python version to the versions installed above](https://github.com/pyenv/pyenv/blob/master/COMMANDS.md#pyenv-global):
+
+```shell
+pyenv global 3.6.8 2.7.15
+```
+
+This will make `python` version 3.6.8, and `python2` will be 2.7.15. This is our recommended configuration.
+
+
+## Using local Python versions
+
+pyenv also allows you to set a local Python version (the version of Python that is available in the current directory):
+
+```shell
+pyenv local 2.7.15
+```
+
+This will create a `.python-version` file in the current directory. When you're in that directory or below it, pyenv will use the version you've specified here as `python`. 
```

Comment:
Do we want to make a note about whether to check in `.python-version` files or if they should be added to the `.gitignore`? Do we have a policy about this yet? (May be a BEWD convo)

Reply:
Definitely a BEWD conversation. I could see major advantages and major disadvantages to letting .python-version being committed.

Reply:
https://github.com/cfpb/cfgov-refresh/blob/master/.gitignore#L75

Reply:
Setting local versions is somewhat invasive to me, and I'd lead toward current policy of excluding .python-version from commits.

Reply:
It's not as invasive as autoenv, but it walks like autoenv.

Reply:
Yeah — I have a visceral reaction against it. *But* it could be a nice way to ensure that we're all using exactly the same python version for particular projects. 

I lean in favor of gitignoring it... but I could see arguments in the alternative. It might be a per-project preference?

Reply:
One sticky thing is patch-level versions. Is it possible to commit a `.python-version` file with just `2.7` or `3.6` in it that will activate any version beginning with those major/minor versions?

Reply:
Unfortunately not:

```
% cat .python-version 
2.7
% python             
pyenv: version `2.7' is not installed (set by /Users/bartonw/test/.python-version)
```

Reply:
What about adding a note that you *may* want to add `.python-version` to `.gitignore` and avoid committing it, and maybe adding it to the [sample `.gitignore` file in this repo](https://github.com/cfpb/development/blob/master/.gitignore)?

Reply:
> Unfortunately not:
> 
> ```
> % cat .python-version 
> 2.7
> % python             
> pyenv: version `2.7' is not installed (set by /Users/bartonw/test/.python-version)
> ```

To me, that makes committing them a definite no-go.

> What about adding a note that you _may_ want to add `.python-version` to `.gitignore` and avoid committing it, and maybe adding it to the [sample `.gitignore` file in this repo](https://github.com/cfpb/development/blob/master/.gitignore)?

And if we want to discourage committing it, I would add it to the sample and use a stronger word than "may".

Reply:
Ouch! Another vote for adding it to our sample `.gitignore` and a stronger warning against committing `.python_version` files.

Reply:
Alrighty, adding.

Reply:
Added in 10dd435.